app/lib/shared/memcache.dart

```diff
@@ -47,6 +47,24 @@ class SimpleMemcache<T> {
     }
   }
 
+  Future<List<int>> getBytes(T key) async {
+    try {
+      return await _memcache.get(_key(key), asBinary: true);
+    } catch (e, st) {
+      _logger.severe('Error accessing memcache:', e, st);
+    }
+    _logger.fine('Couldn\'t find memcache entry for $key');
+    return null;
+  }
+
+  Future setBytes(T key, List<int> content) async {
+    try {
+      await _memcache.set(_key(key), content, expiration: _expiration);
+    } catch (e, st) {
+      _logger.warning('Couldn\'t set memcache entry for $key', e, st);
+    }
+  }
+
   Future invalidate(T key) async {
     try {
       await _memcache.remove(_key(key));
```

Comment:
I think you can move the invalidation logging here. It is beneficial to have the cache invalidation log messages and they happen almost never (only on new package uploads).